cli/internal/terraform/terraform/azure/main.tf

```diff
@@ -41,6 +41,12 @@ resource "random_password" "initSecret" {
   override_special = "_%@"
 }
 
+resource "azurerm_attestation_provider" "attestation_provider" {
```

Comment:
> Should we gate creation of the MAA policy provider behind actually having that attestation mode enabled in the config?

Yes, we should.

Reply:
@daniel-weisse should I just use pass an extra variable into the Terraform call and then [let Terraform create the resource conditionally](https://developer.hashicorp.com/terraform/language/expressions/conditionals) depending on that variable? Or do we have a best-practice way to go here?

Reply:
>should I just use pass an extra variable into the Terraform call

Yes.
Configuration will happen through the already existing `EnforceIDKeyDigest` key in the config.